### PR TITLE
Quality check escaped ampersand

### LINKED ISSUE #8948: Improve Equality check for Journal abbreviations
I have a question regarding the use of ampersands `&`, I often come across `\&` in my `.bib` files when an ampersand is used, JabRef doesn't seem to recognize that journaltitle if the name is not exactly the same. Do we need to add entries for all occurences of `&` and `\&` and `and` as well?

_Originally posted by @jhossbach in https://github.com/JabRef/abbrv.jabref.org/issues/100#issuecomment-1129177022_

### PR DESCRIPTION
**Pull request by deybis17**
Fixes for  #8948 
- Unscaped ampersands can be found and displayed on the check integrity window
- Tests added for escaped ampersand

```[tasklist]
### Compulsory checks
- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [x] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.
```

### CODE REVIEW

**Comment by Siedlerchr** [comment]
Please have a look at the checkstlye issues. Just go to the Files changes tab and reviewdog will show you the offending issues

File: src/main/resources/l10n/JabRef_de.properties
Comment (calixtus) [inline src/main/resources/l10n/JabRef_de.properties:1]: Just a note: Although this should work both ways, translations should normally only be done with crowdin. Unified and tested common workflows lead to less errors.

Comment (Siedlerchr) [reply src/main/resources/l10n/JabRef_de.properties:1]: Yep, Crowdin will overwrite these on the next update

Comment (Zylence) [reply src/main/resources/l10n/JabRef_de.properties:1]: Thank you for the information. We will make sure to do it the proper way next time.

**Review by Siedlerchr — APPROVED** [review]
can you please add a changelog something like
 " We added a new Integrity check for unscaped ampersands "

**Comment by Zylence** [comment]
> can you please add a changelog something like " We added a new Integrity check for unscaped ampersands "

Okay, done :)

**Comment by Siedlerchr** [comment]
Please merge the latest main into it (I just fixed an issue related from another pr) so that the unit tests are green again

**Review by Siedlerchr — APPROVED** [review]

File: src/test/java/org/jabref/logic/integrity/AmpersandCheckerTest.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    void rejectsDisallowedInputs(String expectedMessage, Field field, String value) {
+        entry.setField(field, value);
+        assertEquals(List.of(new IntegrityMessage(expectedMessage, entry, field)), checker.check(entry));
+    }
+
+    private static Stream<Arguments> provideUnacceptedInputs() {
+        return Stream.of(
+                Arguments.of("Found 1 unescaped '&'", StandardField.SUBTITLE, "A single &"),
+                Arguments.of("Found 2 unescaped '&'", StandardField.ABSTRACT, "Multiple \\\\& not properly & escaped"),
+                Arguments.of("Found 1 unescaped '&'", StandardField.AUTHOR, "To many backslashes \\\\&"),
```
Comment (koppor) [inline src/test/java/org/jabref/logic/integrity/AmpersandCheckerTest.java:52]: Small typo:

```suggestion
                Arguments.of("Found 1 unescaped '&'", StandardField.AUTHOR, "Too many backslashes \\\\&"),
```

File: src/test/java/org/jabref/logic/integrity/AmpersandCheckerTest.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                Arguments.of("Found 2 unescaped '&'", StandardField.LABEL, "\\\\\\\\& Multiple times \\\\& multiple backslashes")
+        );
+    }
+
+    @Test
+    void entryWithEscapedAndUnescapedAmpersand() {
+        entry.setField(StandardField.TITLE, "Jack \\& Jill & more");
+        assertEquals(List.of(new IntegrityMessage("Found 1 unescaped '&'", entry, StandardField.TITLE)), checker.check(entry));
+    }
```
Comment (koppor) [inline src/test/java/org/jabref/logic/integrity/AmpersandCheckerTest.java:61]: Can you integrate that in `rejectsDisallowedInputs`? I think, there is no additional logic here in this test case.

File: src/test/java/org/jabref/logic/integrity/AmpersandCheckerTest.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+    @Test
+    void entryWithMultipleEscapedAndUnescapedAmpersands() {
+        entry.setField(StandardField.AFTERWORD, "May the force be with you & live long \\\\& prosper \\& to infinity \\\\\\& beyond & assemble \\\\\\\\& excelsior!");
+        assertEquals(List.of(new IntegrityMessage("Found 4 unescaped '&'", entry, StandardField.AFTERWORD)), checker.check(entry));
+    }
+}
```
Comment (koppor) [inline src/test/java/org/jabref/logic/integrity/AmpersandCheckerTest.java:68]: Can you integrate that in `rejectsDisallowedInputs`? I think, there is no additional logic here in this test case.

**Review by koppor — CHANGES_REQUESTED** [review]
Thank you for the PR! The implementation covered more cases than I thought of. Nice and clean test cases.

Neverthless, I have some nitpicks regarding the tests.

**Comment by koppor** [comment]
@Siedlerchr already merged the PR. Maybe, you can do a follow-up PR @deybis17?

File: src/test/java/org/jabref/logic/integrity/AmpersandCheckerTest.java (hunk 3 above)
Comment (Zylence) [reply src/test/java/org/jabref/logic/integrity/AmpersandCheckerTest.java:68]: Sure, we can do that. The reasoning behind having it separated was that it basically combined the test cases from before, containing both escaped and unescaped ampersands (in the parametrized tests there was no mixing of the two).

**Comment by koppor** [comment]
This refs https://github.com/JabRef/jabref/issues/8712.

**Pull request merged**